### PR TITLE
Better large file error, fixes #2486

### LINKED ISSUE #2486: File size limit error message should include size of the limit
Since we don't document the file size limit for uploads anywhere, it'd be nice if the error message when an upload is over the limit includes what the limit is, especially considering that the limits might be different for various projects.

Currently the message is something like:

```
HTTPError: 400 Client Error: File too large. for url: https://upload.pypi.org/legacy/
```

It'd be nice if this was something like:

```
HTTPError: 400 Client Error: File too large (limit for <project_name> is <limit>) for url: https://upload.pypi.org/legacy/
```

### PR DESCRIPTION
**Pull request by Miserlou**
First stab.

### CODE REVIEW

**Comment by alexmuller** [comment]
It looks like the linter is still a little bit unhappy. There's a single trailing space on the `HTTPBadRequest` line. It might be better to format it like this to get around the indentation problem? 

```
                    raise _exc_with_message(
                        HTTPBadRequest,
                        "File too large. Limit for project {} is {}.".format(
                            project.name, file_size_limit))
```

File: warehouse/forklift/legacy.py [hunk 1]
```diff
@@ -864,7 +864,9 @@ def file_upload(request):
                     lambda: request.POST["content"].file.read(8096), b""):
                 file_size += len(chunk)
                 if file_size > file_size_limit:
-                    raise _exc_with_message(HTTPBadRequest, "File too large.")
+                    raise _exc_with_message(HTTPBadRequest, 
+                       "File too large. Limit for project {} is {}.".format(
+                           project.name, file_size_limit))
```
Comment (di) [inline warehouse/forklift/legacy.py]: Note that `file_size_limit` is in bytes, so this is going to output something like:

> File is too large. Limit for project lambda_packages is 62914560.

This should probably be in MB, with a unit. It'd be nice if the project name was in quotes as well.

**Comment by brainwane** [comment]
Hi @Miserlou -- how's this going?

**Comment by di** [comment]
(I pinged @Miserlou elsewhere and he said he'll be able to wrap this up)

**Comment by brainwane** [comment]
In a Warehouse developers' [meeting today](https://wiki.python.org/psf/PackagingWG/2017-12-19-Warehouse) we decided a fix to #2486 belongs in [the first milestone, the MVP for package maintainers](https://github.com/pypa/warehouse/milestone/8). So we're looking forward to getting an updated PR and reviewing it! Thanks.

**Comment by di** [comment]
Closing this since it's in our first milestone, so someone else can have a chance at wrapping it up sooner than later. Thanks for giving it a shot @Miserlou!

**Pull request closed without merging**